### PR TITLE
Frameworks package.json fixes

### PR DESCRIPTION
**Pull request by jamesdaniels**
* Remove scripts from package.json
* Convert bundled deps to deps
* Make engine picking less bad

### CODE REVIEW

File: src/frameworks/index.ts [hunk 1]
```diff
@@ -120,6 +122,12 @@ export const ALLOWED_SSR_REGIONS = [
   { name: "asia-east1 (Taiwan)", value: "asia-east1" },
 ];
 
+function conjoinOptions(opts: any[], conjunction: string, seperator: string = ","): string {
+  if (opts.length === 1) return opts[0].toString();
+  const lastElement = opts.slice(-1)[0];
+  return `${opts.slice(0, -1).join(`${seperator} `)}, ${conjunction} ${lastElement}`;
```
Comment (jamesdaniels) [inline src/frameworks/index.ts:128]: use separator for this last comma

Comment (leoortizz) [inline src/frameworks/index.ts:128]: just a typo 😛
```suggestion
function conjoinOptions(opts: any[], conjunction: string, separator: string = ","): string {
  if (opts.length === 1) return opts[0].toString();
  const lastElement = opts.slice(-1)[0];
  return `${opts.slice(0, -1).join(`${separator} `)}, ${conjunction} ${lastElement}`;
```

File: src/frameworks/index.ts [hunk 2]
```diff
@@ -346,7 +354,7 @@ export async function prepareFrameworks(
     }
     const ssrRegion = frameworksBackend?.region ?? DEFAULT_REGION;
     if (!allowedRegionsValues.includes(ssrRegion)) {
-      const validRegions = allowedRegionsValues.join(", ");
+      const validRegions = conjoinOptions(allowedRegionsValues, "or");
```
Comment (chalosalvador) [inline src/frameworks/index.ts:357]:
```suggestion
      const validRegions = conjoinOptions(allowedRegionsValues, "and");
```

File: src/frameworks/index.ts [hunk 3]
```diff
@@ -542,7 +550,20 @@ export async function prepareFrameworks(
...
+        logWarning(
+          `This integration expects Node version ${conjoinOptions(
+            VALID_ENGINES.node,
+            "or"
+          )}. You're running version ${NODE_VERSION}, problems may be encountered.`
+        );
+      }
+      packageJson.engines.node ||= engine.toString();
+      delete packageJson.scripts;
+      delete packageJson.devDependencies;
```
Comment (jamesdaniels) [inline src/frameworks/index.ts:565]: we're already removing devDeps above, we should consolidate

File: src/frameworks/index.ts [hunk 4]
```diff
@@ -542,7 +550,20 @@ export async function prepareFrameworks(
...
+      delete packageJson.bundledDependencies;
```
Comment (jamesdaniels) [inline src/frameworks/index.ts:566]: warn and convert to deps

**Review by leoortizz — APPROVED** [review]
LGTM. I added tests for `conjoinOptions` and fixed some issues with the returned string

**Pull request merged**